File: src/main/java/masters/flexilag/HaxelibLagChecker.java
package masters.flexilag;

import masters.libiostudy.Version;

/**
 * @author ______
 */
public class HaxelibLagChecker implements LagChecker {

    @Override
    public MatcherResult matches(Version version, String classification, String declaration) {
        return MatcherResult.NOT_SUPPORTED;
    }

    @Override
    public Declaration getDeclaration(String classification, String declaration) {
        return null;
    }

}

File: src/main/java/masters/flexilag/HaxelibLagChecker.kt
package masters.flexilag

import masters.libiostudy.Version

/**
 * @author ______
 */
class HaxelibLagChecker : LagChecker {
    override fun getDeclaration(classification: String, declaration: String): Declaration {
        return if (declaration == "*")
            Declaration.getAny()
        else
            Declaration(Version.create(declaration), Version.create(declaration))
    }
}